v03_pipeline/lib/misc/sample_entries.py

```diff
@@ -0,0 +1,56 @@
+import hail as hl
+
+
+def _empty_entries(ht: hl.Table) -> hl.StructExpression:
+    first_sample_ids_row = ht.aggregate(hl.agg.take(ht.entries.sample_id, 1))
+    if not len(first_sample_ids_row):
+        return hl.empty_array(ht.entries.dtype.element_type)
+    return [
+        hl.Struct(
+            **{
+                k: hl.missing(v)
+                for k, v in ht.entries.dtype.element_type.items()
+                if k != 'sample_id'
+            },
+            sample_id=sample_id,
+        )
+        for sample_id in first_sample_ids_row[0]
+    ]
+
+
+def globalize_sample_ids(ht: hl.Table) -> hl.Table:
+    ht = ht.annotate_globals(
+        sample_ids=ht.aggregate(hl.agg.take(ht.entries.sample_id, 1)[0]),
```

Comment:
wouldn't this only take the first sample_id?

Reply:
`hl.agg.take` returns the list as a list....

e.g.

```
In [8]: ht.show()
+-------+-----------------------------------------+
|    id | entries                                 |
+-------+-----------------------------------------+
| int32 | array<struct{a: int32, sample_id: str}> |
+-------+-----------------------------------------+
|     0 | [(1,"a"),(2,"c"),(1,"e"),(2,"f")]       |
|     1 | [(2,"a"),(3,"c"),(4,"e"),(5,"f")]       |
+-------+-----------------------------------------+

In [9]: ht.aggregate(hl.agg.take(ht.entries.sample_id, 1))
2023-06-05 14:27:12.567 Hail: INFO: Coerced sorted dataset
Out[9]: [['a', 'c', 'e', 'f']]

In [10]: ht.aggregate(hl.agg.take(ht.entries.sample_id, 1))[0]
2023-06-05 14:27:51.531 Hail: INFO: Coerced sorted dataset
Out[10]: ['a', 'c', 'e', 'f']
```